From 3945547e161535f0834efdf371ff6a55abd01e21 Mon Sep 17 00:00:00 2001
From: Yufei Gu <yufei.gu@cloudera.com>
Date: Fri, 21 Apr 2017 11:25:25 -0700
Subject: [PATCH 2375/2848] YARN-6453. fairscheduler-statedump.log gets
 generated regardless of service (Contributed by
 Yufei Gu via Daniel Templeton)

(cherry picked from commit e74ca7216aecc998d277e5ca1f4d0abae43c7e31)
(cherry picked from commit d77137a03f89faf43cdf4702e9d1c58febdf130a)

Conflicts:
	hadoop-common-project/hadoop-common/src/main/conf/log4j.properties
	hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/markdown/FairScheduler.md

Change-Id: I831d1ab1a2d8a7c53b0414d1e34986fea28fbc12
---
 .../hadoop-common/src/main/conf/log4j.properties   |   22 ++++++++++++--------
 .../src/site/apt/FairScheduler.apt.vm              |   12 +++++++++++
 2 files changed, 25 insertions(+), 9 deletions(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/conf/log4j.properties b/hadoop-common-project/hadoop-common/src/main/conf/log4j.properties
index 6b7ee8f..86fe8df 100644
--- a/hadoop-common-project/hadoop-common/src/main/conf/log4j.properties
+++ b/hadoop-common-project/hadoop-common/src/main/conf/log4j.properties
@@ -269,12 +269,16 @@ log4j.appender.RMSUMMARY.layout.ConversionPattern=%d{ISO8601} %p %c{2}: %m%n
 #log4j.appender.nodemanagerrequestlog.Filename=${hadoop.log.dir}/jetty-nodemanager-yyyy_mm_dd.log
 #log4j.appender.nodemanagerrequestlog.RetainDays=3
 
-# Fair scheduler requests log on state dump
-log4j.logger.org.apache.hadoop.yarn.server.resourcemanager.scheduler.fair.FairScheduler.statedump=DEBUG,FSLOGGER
-log4j.additivity.org.apache.hadoop.yarn.server.resourcemanager.scheduler.fair.FairScheduler.statedump=false
-log4j.appender.FSLOGGER=org.apache.log4j.RollingFileAppender
-log4j.appender.FSLOGGER.File=${hadoop.log.dir}/fairscheduler-statedump.log
-log4j.appender.FSLOGGER.layout=org.apache.log4j.PatternLayout
-log4j.appender.FSLOGGER.layout.ConversionPattern=%d{ISO8601} %p %c: %m%n
-log4j.appender.FSLOGGER.MaxFileSize=${hadoop.log.maxfilesize}
-log4j.appender.FSLOGGER.MaxBackupIndex=${hadoop.log.maxbackupindex}
+#
+# Fair scheduler state dump
+#
+# Use following logger to dump the state to a separate file
+
+#log4j.logger.org.apache.hadoop.yarn.server.resourcemanager.scheduler.fair.FairScheduler.statedump=DEBUG,FSSTATEDUMP
+#log4j.additivity.org.apache.hadoop.yarn.server.resourcemanager.scheduler.fair.FairScheduler.statedump=false
+#log4j.appender.FSSTATEDUMP=org.apache.log4j.RollingFileAppender
+#log4j.appender.FSSTATEDUMP.File=${hadoop.log.dir}/fairscheduler-statedump.log
+#log4j.appender.FSSTATEDUMP.layout=org.apache.log4j.PatternLayout
+#log4j.appender.FSSTATEDUMP.layout.ConversionPattern=%d{ISO8601} %p %c: %m%n
+#log4j.appender.FSSTATEDUMP.MaxFileSize=${hadoop.log.maxfilesize}
+#log4j.appender.FSSTATEDUMP.MaxBackupIndex=${hadoop.log.maxbackupindex}
diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/apt/FairScheduler.apt.vm b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/apt/FairScheduler.apt.vm
index 635f2a9..b895217 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/apt/FairScheduler.apt.vm
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-site/src/site/apt/FairScheduler.apt.vm
@@ -491,3 +491,15 @@ Moving applications between queues
   of determining fairness. An attempt to move an application to a queue will
   fail if the addition of the app's resources to that queue would violate the
   its maxRunningApps or maxResources constraints.
+
+Dumping Fair Scheduler state
+
+ Fair Scheduler is able to dump its state periodically. It is disabled by
+ default. The administrator can enable it by setting
+ org.apache.hadoop.yarn.server.resourcemanager.scheduler.fair.FairScheduler
+ logging level to DEBUG.
+
+ Fair Scheduler logs go to the Resource Manager log file by default.
+ Fair Scheduler state dumps can potentially generate large amount of log data.
+ Uncomment the "Fair scheduler state dump" section in log4j.properties to dump
+ the state into a separate file.
\ No newline at end of file
-- 
1.7.9.5

